src/views/Give/DepositRow.tsx

```diff
@@ -0,0 +1,182 @@
+import data from "./projects.json";
+import { Project } from "../../components/GiveProject/project.type";
+import useMediaQuery from "@material-ui/core/useMediaQuery";
+import { ManageDonationModal, WithdrawSubmitCallback } from "./ManageDonationModal";
+import { SubmitCallback } from "src/views/Give/Interfaces";
+import { error } from "../../slices/MessagesSlice";
+import { t } from "@lingui/macro";
+import { useWeb3Context } from "src/hooks/web3Context";
+import { NetworkId } from "src/constants";
+import { EnvHelper } from "src/helpers/Environment";
+import { ACTION_GIVE_EDIT, ACTION_GIVE_WITHDRAW, changeGive, changeMockGive } from "../../slices/GiveThunk";
+import "./give.scss";
+
+interface IUserDonationInfo {
+  date: string;
+  deposit: string;
+  recipient: string;
+  yieldDonated: string;
+}
+
+interface DepositRowProps {
+  depositObject: IUserDonationInfo;
+}
+
+export const DepositTableRow = ({ depositObject }: DepositRowProps) => {
+  const dispatch = useDispatch();
+  const { provider, address, networkId } = useWeb3Context();
+  const { projects } = data;
+  const projectMap = new Map(projects.map(i => [i.wallet, i] as [string, Project]));
+  const [isManageModalOpen, setIsManageModalOpen] = useState(false);
+  const [isEditModalOpen, setIsEditModalOpen] = useState(false);
+  const [isWithdrawModalOpen, setIsWithdrawModalOpen] = useState(false);
+  const isSmallScreen = useMediaQuery("(max-width: 600px)");
+
+  const getRecipientTitle = (address: string): string => {
+    const project = projectMap.get(address);
+    if (!project) return "Custom Recipient";
+
+    if (!project.owner) return project.title;
+
+    return project.owner + " - " + project.title;
+  };
+
+  const handleManageModalCancel = () => {
+    setIsManageModalOpen(false);
+  };
+
+  const handleEditModalSubmit: SubmitCallback = async (
+    walletAddress,
+    eventSource,
+    depositAmount,
+    depositAmountDiff,
+  ) => {
+    if (!depositAmountDiff) {
+      return dispatch(error(t`Please enter a value!`));
+    }
+
+    if (depositAmountDiff.isEqualTo(new BigNumber(0))) return;
+
+    // If reducing the amount of deposit, withdraw
+    if (networkId === NetworkId.TESTNET_RINKEBY && EnvHelper.isMockSohmEnabled(location.search)) {
+      await dispatch(
+        changeMockGive({
+          action: ACTION_GIVE_EDIT,
+          value: depositAmountDiff.toFixed(),
+          recipient: walletAddress,
+          provider,
+          address,
+          networkID: networkId,
+          version2: false,
+          rebase: false,
+          eventSource,
+        }),
+      );
+    } else {
+      await dispatch(
+        changeGive({
+          action: ACTION_GIVE_EDIT,
+          value: depositAmountDiff.toFixed(),
+          recipient: walletAddress,
+          provider,
+          address,
+          networkID: networkId,
+          version2: false,
+          rebase: false,
+          eventSource,
+        }),
+      );
+    }
+
+    setIsEditModalOpen(false);
```

Comment:
Is this correct? Or should it be manage modal? As isEditModalOpen is never referenced.

Reply:
not correct. will change

Reply:
fixed